protocol/deposits-interop.md

```diff
@@ -0,0 +1,189 @@
+
+$$f:A→\{b1,b2\}$$
+
+A deposit transaction event can be mapped to either a valid deposit transaction or a noop deposit transaction.
+
+A new invariant called the "sequencing window invariant" is introduced that enforces a delay for deposit
+transactions that trigger executing messages.
+
+## Derivation Pipeline Execution
+
+The derivation pipeline is updated to include EVM execution of the deposit transactions. A forking EVM provider backed
+by L2 tip state is used. This enables simulation of the deposit transactions before they are included in L2. If the deposit
+transaction fails the sequencing window invariant or creates an invalid executing message, it is mapped into a noop deposit
+transaction.
+
+A noop deposit transaction:
+- keeps its sourcehash for tx hash uniqueness
+- keeps its from, so the nonce can increment
+- keeps its to, in case its also depositing `ether`
+- keeps its mint, in case its depositing `ether`
+- gas is set to 0
+- data is set to empty
+
+This should hold the invariant that the `ether` is always minted when sent into the `OptimismPortal` but prevents
+any additional execution on top of that.
+
+In pseudocode:
+
+```python
+txs = []
+evm = prepare_evm(eth)
+for i, tx in enumerate(payload.transactions):
+  if tx.type() != DEPOSIT_TYPE:
+    tx.append(tx)
+    continue
+  logs = evm.call(tx.to_message())
+  for log in logs:
+    if is_executing_message(log):
+      if sequencing_window_invariant(log) == False or is_valid_message(log) == False:
+        tx.gas = 0
+        tx.data = ""
+        txs.append(tx)
+        break
+```
+
+If static analysis was possible, then no execution would be required to derive the information required to check the
+validity of the executing messages. The same two checks `sequencing_window_invariant` and `is_valid_message` could
+be applied but solely with static analysis instead of needing to execute to get the executing message.
+The definition of `sequencing_window_invariant` is defined below and `is_valid_message` is defined in the specs.
+
+Note that only the deposits are simulated as part of the derivation pipeline, not any transactions that are sent
+directly to the sequencer. This means that there is a guaranteed worst case in the amount of gas for this execution.
+
+## Sequencing Window Invariant
+
+With or without the only EOA invariant, the sequencing window invariant needs to be defined to prevent
+race conditions with deposit transactions and the sequencer having a slow p2p network. Without this invariant,
+it would be possible to trigger reorgs by sending a deposit transaction that references an initiating message
+that is at the tip of a remote chain. If the sequencer doesn't immediately always have the remote chain, it
+could result in reorgs.
+
+The definition of the sequencing window invariant is as follows:
+
+```
+require(identifier.timestamp + sequencer_window <= block.timestamp)
+```
```

Comment:
This requires either:
- standard sequencer window for all chains
- sequencer window of remote chains is legible to the derivation pipeline